modules/entrezdirect/esummary/main.nf

```diff
@@ -0,0 +1,39 @@
+process ENTREZDIRECT_ESUMMARY {
+    tag "$meta.id"
+    label 'process_low'
+
+    conda (params.enable_conda ? "bioconda::entrez-direct=16.2" : null)
+    container "${ workflow.containerEngine == 'singularity' && !task.ext.singularity_pull_docker_container ?
+        'https://depot.galaxyproject.org/singularity/entrez-direct:16.2--he881be0_1':
+        'quay.io/biocontainers/entrez-direct:16.2--he881be0_1' }"
+
+    input:
+    tuple val(meta)
+    val uid
+    val database
+
+    output:
+    tuple val(meta), path("*.xml"), emit: xml_esummary
+    path "versions.yml"           , emit: versions
+
+    when:
+    task.ext.when == null || task.ext.when
+
+    script:
+    def args = task.ext.args ?: ''
+    def prefix = task.ext.prefix ?: "${meta.id}"
+    def std_input = uid ? "-id ${uid}" : null
```

Comment:
```suggestion
    if ( uid && uids_file )  error "Cannot supply both a uid and file of uids at the same time. Check input"
    if ( !uid && !uids_file )  error "Must supply either a uid or uids file. Check input"
    def input = uid ? "-id ${uid}" : "-file ${uids_file}"
```

Reply:
Note this is maybe not the most elegant, but best I can do in a rush... I'm not sure if there is a way to dynamically load a file as a value or file (I don't think there is in NXF).

Reply:
Actually could probably condense the input check by combining the conditions into one error

Reply:
Thanks. I willl try it and include it in the next commit.